File: Data/Project Classes/FramingInvoice.cs
namespace SJC_Portal.Data
{
    public class FramingInvoice
    {
        public int? InvoiceNumber { get; set; }
        public string? InvoiceDate { get; set; }
        public IEnumerable<PairList>? Buildings { get; set; }
        public IEnumerable<PairList>? ToBeInvoiced { get; set; }
        public double? InvoiceCharge { get; set; }
        public double? CraneCharge { get; set; }

    }
}

File: wwwroot/js/ProjectClass/FramingInvoice.js
import { PairList } from "./PairDetails.js";
export class FramingInvoice {
    InvoiceNumber;
    InvoiceDate;
    Buildings;
    ToBeInvoiced;
    InvoiceCharge;
    CraneCharge;

    constructor() {
        this.InvoiceNumber = "";
        this.InvoiceDate = "";
        this.Buildings = [new PairList()];
        this.Buildings.pop();
        this.ToBeInvoiced = [new PairList()];
        this.ToBeInvoiced.pop();
        this.InvoiceCharge = 0;
        this.CraneCharge = 0;
    }
}